File: DSA-python/hasmap-question/leet160.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getIntersectionNode(self, headA, headB):
        """
        :type head1, head1: ListNode
        :rtype: ListNode
        """
        store = set()
        curra = headA
        curr = headB
        while(curr):
            store.add(curr)
            curr = curr.next
        while(curra):
            if curra in store:
                return curra
            curra = curra.next
        return None

        

File: DSA-js/hasmap-question/leet160.js
// checking the interesection of the twolinked list at the same time
var getIntersectionNode = function(headA, headB) {
    let curra = headA
    let currb = headB
    let store = new Set();
    while(currb)
    {
        store.add(currb);
        currb = currb.next;
    }
    while(curra)
    {
        if(store.has(curra))
        {
            return curra;
        }
        curra = curra.next;
    }
    return null;
};